conan/conan.go

```diff
@@ -91,14 +93,33 @@ func (c ConanGenerator) checkForDuplicates(purls []packageurl.PackageURL) []cona
 }
 
 func (c ConanGenerator) writeConanFile(purls []conanPurlInfo) {
-    var data strings.Builder
-    data.WriteString("[requires]")
-    data.WriteString(newline)
-    for _, purl := range purls[:] {
-        data.WriteString(purl.name)
-        data.WriteString("/")
-        data.WriteString(purl.version)
-        data.WriteString(newline)
+    file, err := os.Create(c.filepath)
+    if err != nil {
+        c.logger.Error(err)
+        return
+    }
+    defer file.Close()
+
+    header := "[requires]" + newline
+    n, err := file.WriteString(header)
```

Comment:
104-112 are BASICALLY identical (so you could make a private method for doing the damn thing). Buyers choice though!